pkg/reconciler/v1alpha1/revision/revision_test.go

```diff
@@ -740,37 +836,27 @@ func TestGlobalResyncOnConfigMapUpdate(t *testing.T) {
 
 			rev := getTestRevision()
 			revClient := servingClient.ServingV1alpha1().Revisions(rev.Namespace)
-			deploymentsClient := kubeClient.Apps().Deployments(rev.Namespace)
 			h := NewHooks()
 
-			h.OnUpdate(&servingClient.Fake, "revisions", func(obj runtime.Object) HookResult {
-				updatedRev := obj.(*v1alpha1.Revision)
-				t.Logf("Revision updated: %v", updatedRev.Name)
-				updatedDeployment, err := deploymentsClient.Get(resourcenames.Deployment(updatedRev), metav1.GetOptions{})
-				if err != nil {
-					t.Error(err)
-				}
-
-				got, wasUpdated := test.wasUpdated(test.expected, updatedRev, updatedDeployment)
-
-				if !wasUpdated {
-					t.Logf("No update occurred; expected: %s got: %s", test.expected, got)
-					return HookIncomplete
-				}
-				return HookComplete
-			})
-
 			servingInformer.Start(stopCh)
 			kubeInformer.Start(stopCh)
 			cachingInformer.Start(stopCh)
+			if err := watcher.Start(stopCh); err != nil {
+				t.Fatalf("Failed to start configuration manager: %v", err)
+			}
 
 			servingInformer.WaitForCacheSync(stopCh)
 			kubeInformer.WaitForCacheSync(stopCh)
 			cachingInformer.WaitForCacheSync(stopCh)
 
-			if err := watcher.Start(stopCh); err != nil {
-				t.Fatalf("Failed to start config map watcher: %v", err)
-			}
+			h.OnCreate(&kubeClient.Fake, "deployments", func(obj runtime.Object) HookResult {
```

Comment:
I believe moving this here creates a race

Reply:
Done